File: Leetcode/PowerofFour.rs
// https://leetcode.com/problems/power-of-four

struct Solution {}

impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        let x = (n as f64).log2();
        x.fract() == 0.0 && (x.trunc() as i32) % 2 == 0
    }
}

File: Leetcode/PowerofFour.cpp
// https://leetcode.com/problems/power-of-four

class Solution {
   public:
    bool isPowerOfFour(int n) {
        using i64 = long long;

        if (n <= 0) {
            return false;
        }

        i64 cur = 1;
        while (cur <= n) {
            if (cur == n) {
                return true;
            }
            cur *= 4;
        }

        return false;
    }
};